### PR TITLE
Add TX/RX message direction to socketcan

### PR DESCRIPTION
**Pull request by tysonite**
References #773 and #776

### CODE REVIEW

**Comment by tysonite** [comment]
Don't judge me too much, please. :) My first try to submit something there. I actually got a pointer to implement that looking into [candump tool](https://github.com/linux-can/can-utils/blob/da65fdfe0d1986625ee00af0b56ae17ec132e700/candump.c#L778) and [SocketCAN doc](https://www.kernel.org/doc/Documentation/networking/can.txt) 4.1.7 section.

File: can/interfaces/socketcan/socketcan.py [hunk 1]
```diff
@@ -536,6 +536,7 @@ def capture_message(
     is_remote_transmission_request = bool(can_id & CAN_RTR_FLAG)
     is_error_frame = bool(can_id & CAN_ERR_FLAG)
     is_fd = len(cf) == CANFD_MTU
+    is_tx = bool(msg_flags & socket.MSG_DONTROUTE)
```
Comment (bessman) [inline can/interfaces/socketcan/socketcan.py]: I would add a comment explaining the use of this flag:
`# MSG_DONTROUTE: set when the received frame was created on the local host.`

File: can/interfaces/socketcan/socketcan.py [hunk 2]
```diff
@@ -555,6 +556,7 @@ def capture_message(
         is_remote_frame=is_remote_transmission_request,
         is_error_frame=is_error_frame,
         is_fd=is_fd,
+        is_rx=not is_tx,
```
Comment (bessman) [inline can/interfaces/socketcan/socketcan.py]: I think `is_rx` should follow the same pattern as the other parameters, i.e. `is_rx=is_rx`. So move the `not` to L539 instead.

Comment (tysonite) [reply can/interfaces/socketcan/socketcan.py]: Done

File: can/interfaces/socketcan/socketcan.py (hunk 1 above)
Comment (tysonite) [reply can/interfaces/socketcan/socketcan.py]: Done

**Pull request merged**